CRM/Twingle/Profile.php

```diff
@@ -177,21 +193,137 @@ public function getTransactionID($twingle_id) {
    * Verifies whether the profile is valid (i.e. consistent and not colliding
    * with other profiles).
    *
-   * @throws Exception
+   * @throws \CRM_Twingle_Exceptions_ProfileValidationError
+   * @throws \Civi\Core\Exception\DBQueryException
    *   When the profile could not be successfully validated.
    */
-  public function verifyProfile() {
-    // TODO: check
-    //  data of this profile consistent?
-    //  conflicts with other profiles?
+  public function validate() {
+
+    // Name cannot be empty
+    if (empty($this->getName())) {
+      throw new ProfileValidationError(
+        'name',
+        E::ts('Profile name cannot be empty.'),
+        ProfileValidationError::ERROR_CODE_PROFILE_VALIDATION_FAILED
+      );
+    }
+
+    // Restrict profile names to alphanumeric characters, space and the underscore.
+    $contains_illegal_characters = preg_match("/[^A-Za-z0-9_\s]/", $this->getName());
+    if ($contains_illegal_characters) {
+      throw new ProfileValidationError(
+        'name',
+        E::ts('Only alphanumeric characters, space and the underscore (_) are allowed for profile names.'),
+        ProfileValidationError::ERROR_CODE_PROFILE_VALIDATION_FAILED
+      );
+    }
```

Comment:
I think this restriction is not necessary anymore now we have profile IDs, but let's just add an issue for discussing that.

Reply:
I created an issue #78 for that question.